### PR TITLE
Pass the extra data as a meta object and not literal strings

### PR DESCRIPTION
**Pull request by danshapir**
The context and trace are ignored when formatted by winston if not passed as objects.

### CODE REVIEW

**Comment by gremo** [comment]
I think you should re-do your work because changed files doesn't match. Maybe the problem is my force push?

**Comment by danshapir** [comment]
> I think you should re-do your work because changed files doesn't match. Maybe the problem is my force push?

What doesn't match? I fixed the conflict...

**Comment by gremo** [comment]
> > I think you should re-do your work because changed files doesn't match. Maybe the problem is my force push?
> 
> What doesn't match? I fixed the conflict...

Nope... I can't see what has been changed, it seems only formatting and commas.

**Comment by danshapir** [comment]
Well, look again...

**Comment by gremo** [comment]
You are changing formatting and README.md already changed in https://github.com/gremo/nest-winston/commit/4b25e261d958eb4cd04d4a90469f72fe428e3d49

**Comment by danshapir** [comment]
@gremo you are kidding right?




It's inside an object now.
Just approve the pull request, I'm trying to assist you. If you prefer I can just create another package of my own...

**Comment by gremo** [comment]
I'll try to start a review to make you understand what I'm saying, ok?

File: README.md [hunk 1]
```diff
@@ -80,7 +80,8 @@ export class AppModule {}
 
 The factory might be async, can inject dependencies with `inject` option and import other modules using the `imports` option.
 
-## Use as the main Nest logger
+
+## Use as the main Nest Logger
```
Comment (gremo) [inline README.md]: Why there is a new line here and you changed "logger" to "Logger"?

File: src/winston.providers.ts [hunk 2]
```diff
@@ -1,59 +1,65 @@
...
+import { createLogger, LoggerOptions, Logger } from "winston";
+import {
+  WINSTON_MODULE_OPTIONS,
+  WINSTON_MODULE_PROVIDER,
+  WINSTON_MODULE_NEST_PROVIDER
+} from "./winston.constants";
+import {
+  WinstonModuleAsyncOptions,
+  WinstonModuleOptions
+} from "./winston.interfaces";
```
Comment (gremo) [inline src/winston.providers.ts]: Why reformatting the imports?

File: src/winston.providers.ts [hunk 3]
```diff
@@ -1,59 +1,65 @@
...
 
 class WinstonLogger implements LoggerService {
-  constructor(private readonly logger: Logger) { }
+  constructor(private readonly logger: Logger) {}
```
Comment (gremo) [inline src/winston.providers.ts]: Why removing the blank space?

File: src/winston.providers.ts [hunk 4]
```diff
@@ -1,59 +1,65 @@
...
 
-export function createWinstonAsyncProviders(options: WinstonModuleAsyncOptions): Provider[] {
+export function createWinstonAsyncProviders(
+  options: WinstonModuleAsyncOptions): Provider[] {
   return [
     {
       provide: WINSTON_MODULE_OPTIONS,
       useFactory: options.useFactory,
-      inject: options.inject || [],
+      inject: options.inject || []
```
Comment (gremo) [inline src/winston.providers.ts]: Why removing comma?

File: src/winston.providers.ts [hunk 5]
```diff
@@ -1,59 +1,65 @@
...
     },
     {
       provide: WINSTON_MODULE_PROVIDER,
       useFactory: (loggerOpts: LoggerOptions) => createLogger(loggerOpts),
-      inject: [WINSTON_MODULE_OPTIONS],
+      inject: [WINSTON_MODULE_OPTIONS]
```
Comment (gremo) [inline src/winston.providers.ts]: Why removing comma?

**Comment by danshapir** [comment]
Man, that's the auto prettier formatting tool. I'll be blunt, either you want my assistance, or I'm just creating a new package called nestjs-winston. The only added logic in your package is mine anyway, you just wrapped winston in a module.
If every time I try and assist it will take that long, sorry I'll just prefer doing it alone...

**Comment by gremo** [comment]
You should not pretty the code if not asked. I'll fix by myself, thank you.

**Pull request closed without merging**